src/libclipper/src/query_processor.cpp

```diff
@@ -72,70 +74,74 @@ boost::future<Response> QueryProcessor::predict(Query query) {
   log_info_formatted(LOGGING_TAG_QUERY_PROCESSOR, "Found {} tasks",
                      tasks.size());
 
-  vector<boost::future<Output>> task_futures =
+  vector<folly::Future<Output>> task_futures =
       task_executor_.schedule_predictions(tasks);
   if (task_futures.empty()) {
     default_explanation = "No connected models found for query";
     log_error_formatted(LOGGING_TAG_QUERY_PROCESSOR,
                         "No connected models found for query with id: {}",
                         query_id);
   }
-  boost::future<void> timer_future =
-      timer_system_.set_timer(query.latency_budget_micros_);
-
-  boost::future<void> all_tasks_completed;
-  auto num_completed = std::make_shared<std::atomic<int>>(0);
-  std::tie(all_tasks_completed, task_futures) =
-      future::when_all(std::move(task_futures), num_completed);
-
-  auto completed_flag = std::make_shared<std::atomic_flag>();
-  // Due to some complexities of initializing the atomic_flag in a shared_ptr,
-  // we explicitly set the value of the flag to false after initialization.
-  completed_flag->clear();
-  boost::future<void> response_ready_future;
 
-  std::tie(response_ready_future, all_tasks_completed, timer_future) =
-      future::when_either(std::move(all_tasks_completed),
-                          std::move(timer_future), completed_flag);
+  size_t num_tasks = task_futures.size();
 
-  boost::promise<Response> response_promise;
-  auto response_future = response_promise.get_future();
+  // TODO(czumar): Find a more readable name for std::vector<folly::Unit> and
+  // typedef accordingly
```

Comment:
I don't think this TODO is necessary. It's not a particularly complicated type and I'd rather leave it explicit.